contracts/LimitsChecker.sol

```diff
@@ -0,0 +1,285 @@
+    constructor(
+        address[] memory _setLimitParameterRoleHolders,
+        address[] memory _updateSpentAmountRoleHolders,
+        IBokkyPooBahsDateTimeContract _bokkyPooBahsDateTimeContract
+    ) {
+        for (uint256 i = 0; i < _setLimitParameterRoleHolders.length; i++) {
+            _setupRole(SET_LIMIT_PARAMETERS_ROLE, _setLimitParameterRoleHolders[i]);
+        }
+        for (uint256 i = 0; i < _updateSpentAmountRoleHolders.length; i++) {
+            _setupRole(UPDATE_SPENT_AMOUNT_ROLE, _updateSpentAmountRoleHolders[i]);
+        }
+        bokkyPooBahsDateTimeContract = _bokkyPooBahsDateTimeContract;
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Checks if _payoutAmount is less or equal than may be spent in the period
+    /// @param _payoutAmount Motion total amount
+    /// @param _motionDuration Motion duration - minimal time required to pass before enacting of motion
+    /// @return True if _payoutAmount is less or equal than may be spent in the period
+    function isUnderSpendableBalance(uint256 _payoutAmount, uint256 _motionDuration)
+        external
+        view
+        returns (bool)
+    {
+        if (block.timestamp + _motionDuration >= currentPeriodEndTimestamp) {
+            return _payoutAmount <= limit;
+        } else {
+            return _payoutAmount <= _currentSpendableBalance();
+        }
+    }
+
+    /// @notice Checks if _payoutAmount is less or equal than may be spent,
+    /// @notice updates period if needed and increases the amount spent in the current period
+    function updateSpendableBalance(uint256 _payoutAmount)
+        external
+        onlyRole(UPDATE_SPENT_AMOUNT_ROLE)
+    {
+        _checkPeriodDurationMonths(periodDurationMonths);
+
+        /// At the moment when it is necessary to shift the currentPeriodEndTimestamp
+        /// currentPeriodEndTimestamp takes on a new value and spent is set to zero. Thus begins a new period.
+        if (block.timestamp >= currentPeriodEndTimestamp) {
+            currentPeriodEndTimestamp = uint128(_getPeriodEndFromTimestamp(block.timestamp));
+            spentAmount = 0;
+        }
+
+        require(_payoutAmount <= _currentSpendableBalance(), ERROR_SUM_EXCEEDS_SPENDABLE_BALANCE);
+
+        spentAmount += uint128(_payoutAmount);
+
+        (
+            uint256 _alreadySpentAmount,
+            uint256 _spendableBalance,
+            uint256 _periodStartTimestamp,
+            uint256 _periodEndTimestamp
+        ) = _getCurrentPeriodState();
+
+        emit SpendableAmountChanged(
+            _alreadySpentAmount,
+            _spendableBalance,
+            _periodStartTimestamp,
+            _periodEndTimestamp
+        );
+    }
+
+    /// @notice Returns balance that can be spent in the current period
+    /// @return Balance that can be spent in the current period
+    function currentSpendableBalance() external view returns (uint256) {
```

Comment:
If period advanced and `updateSpendableBalance()` wasn't called in the new period
`currentSpendableBalance` and `getCurrentPeriodState` will return spentAmount and spendableBalance corresponding to the previous period

I think it's OK, but should add a remark to the docstrings

maybe also remove "current" from their names, which might be a bit more misleading 😅

Reply:
added remarks, renamed methods